test/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/VehicleRegistrationNumberControllerSpec.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggagefrontend.controllers
+
+import play.api.test.Helpers._
+import uk.gov.hmrc.merchandiseinbaggagefrontend.views.html.VehicleRegistrationNumberView
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class VehicleRegistrationNumberControllerSpec extends DeclarationJourneyControllerSpec {
+
+  private lazy val controller =
+    new VehicleRegistrationNumberController(
+      controllerComponents, actionBuilder, declarationJourneyRepository, injector.instanceOf[VehicleRegistrationNumberView])
+
+  "onPageLoad" must {
+    val url = routes.VehicleRegistrationNumberController.onPageLoad().url
+    val getRequest = buildGet(url, sessionId)
+
+    behave like anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToLoad(controller, url)
+
+    "return OK and render the view" when {
+      "a declaration has been started" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney)
+
+        val result = controller.onPageLoad()(getRequest)
+
+        status(result) mustEqual OK
+      }
+    }
+
+    "return OK and render the view" when {
+      "a declaration has been started and a value saved" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney.copy(maybeRegistrationNumber = Some("AB51 CDE")))
+
+        val result = controller.onPageLoad()(getRequest)
+
+        status(result) mustEqual OK
+      }
+    }
+  }
+
+  "onSubmit" must {
+    val url = routes.VehicleRegistrationNumberController.onSubmit().url
+    val postRequest = buildPost(url, sessionId)
+
+    behave like anEndpointRequiringASessionIdAndLinkedDeclarationJourneyToUpdate(controller, url)
+
+    "redirect to /check-your-answers" when {
+      "a declaration is started and a valid selection submitted" in {
+        givenADeclarationJourneyIsPersisted(startedDeclarationJourney)
+
+        val request = postRequest.withFormUrlEncodedBody(("value", "AB51 CDE"))
+
+        val result = controller.onSubmit()(request)
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).get mustEqual routes.CheckYourAnswersController.onPageLoad().toString
+
+        startedDeclarationJourney.maybeRegistrationNumber mustBe None
+        declarationJourneyRepository
```

Comment:
I think we have repo test already so maybe asserting the persistence mechanisms here are redundant, would easier simply by testing that you do call the repository - hope makes sense